src/location.rs

```diff
@@ -1,10 +1,43 @@
 use bevy::prelude::*;
+use crate::direction::{Direction};
 
-pub type XCoord = u8;
-pub type YCoord = u8;
+type Coord = u8;
 
-#[derive(Component, Default, Debug)]
+const MAX_X: Coord = 15;
+const MAX_Y: Coord = 8;
+
+#[derive(Component, Debug)]
 pub struct Location {
-    x: XCoord,
-    y: YCoord,
+    x: Coord,
+    y: Coord,
+}
+
+impl Location {
+    pub fn new() -> Self {
```

Comment:
As a rust style topic, I like default for this. I liked your i8 implementation too, so that (0,0) could be the centre of any size world, and world could be a torus or a bounded rect - I think bounds could be a separate system which would encapsulate that design decision.